pkg/api/norman/store/workload/workload_store.go

```diff
@@ -116,6 +118,22 @@ func (s *CustomizeStore) validateStatefulSetVolume(schema *types.Schema, data ma
 	return nil
 }
 
+// validateNoDeprecatedEnvVars validates that the now deprecated fields `environment` and `environmentFrom` are not being used.
+// The fields were deprecated in favor of k8s native types for being unable to preserve the order of the variables and break interpolation.
+func (s *CustomizeStore) validateNoDeprecatedEnvVars(data map[string]interface{}) error {
+	if containers, _ := values.GetSlice(data, "containers"); len(containers) > 0 {
+		for _, c := range containers {
+			if !convert.IsAPIObjectEmpty(values.GetValueN(c, "environment")) {
+				return httperror.NewAPIError(httperror.InvalidOption, "field `environment` is deprecated, please use Kubernetes native types for the container's environment variables")
+			}
+			if !convert.IsAPIObjectEmpty(values.GetValueN(c, "environmentFrom")) {
+				return httperror.NewAPIError(httperror.InvalidOption, "field `environmentFrom` is deprecated, please use Kubernetes native types for the container's environment variables")
```

Comment:
I used `httperror.InvalidOption` (Code 422) for this because I couldn't find anything more suitabled [here](https://github.com/rancher/norman/blob/8e6ffc77a819d067456dcec666553daf929eba33/httperror/error.go#L8-L34), but please let me know if there is another code more suitable for this case and I'll update this.